src/GADM.jl

```diff
@@ -150,4 +159,42 @@ julia> coordinates("VAT") # Returns a deep array of Vatican city
 """
 coordinates(country, subregions...) = GeoInterface.coordinates(get(country, subregions...))
 
+"""
+    table(country, subregions...)  
+
+Returns named tuples of the region required and its children  
+Input: ISO 3166 Alpha 3 Country Code, and further full official names of subdivisions  
+"""
+function table(country, subregions...) 
+    data = getdataset(country)
+    nlayers = ArchGDAL.nlayer(data)
+
+    function filterlayer(layer, key, value, all=false)
+        table = ArchGDAL.Table(layer)
+        filtered_rows = []
+        for row in Tables.rows(table)
+            field = row[Symbol(key)]
+            if all || occursin(lowercase(value), lowercase(field))
+                push!(filtered_rows, row)
+            end
+        end
+        return filtered_rows
+    end
+    
+    parent_level = length(subregions)
+    parent_level >= nlayers && throw(ArgumentError("more subregions provided than actual")) 
+    parent_name = isempty(subregions) ? "" : last(subregions)
+    parent_layer = getlayer(data, parent_level)
+    result = filterlayer(parent_layer, "NAME_$(parent_level)", parent_name, iszero(parent_level))
+    isempty(result) && throw(ArgumentError("could not find required region"))
+    parent = first(result)
+    
+    children_level = parent_level + 1
+    children_level == nlayers && return (parent, Tables.rowtable([]))
+    children_layer = getlayer(data, children_level)
+    children = filterlayer(children_layer, "NAME_$(parent_level)", parent_name, iszero(parent_level))
+
+    return (parent, Tables.rowtable(children))
```

Comment:
I think the return type should just be a table with the parent included. We don't need to split it?

Reply:
They both are part of different layers so they are meant to be accessed separately.

Reply:
I probably missed something, but the original idea was about retrieving children of a region as a table, right? In case the region had no children, an empty table could be returned. I don't see the point of retrieving the parent in the same query. For that we already have `get`?

Reply:
Let's set a quick call to align this detail tomorrow. I think we may need to consider those options you suggested to the `get` function like `children=true`.

Reply:
@juliohm sorry for the confusion. Here's a detailed explanation of what both of them do:

`table`: Used to get named tuples(metadata + geometric data) of the region requested as well as a table of its children(meta+geom). When there are no children, the table is empty. I return an empty table to keep the API consistent. The user should expect a named tuple and array of named tuples always.

`get`: Used to get just the geometric data of the requested region. (No metadata or children provided)